submissions/kovalov/frogger/app.js

```diff
@@ -0,0 +1,120 @@
+class Enemy {
+  constructor(x, y, speedX, fieldWidth, player) {
+    this.sprite = 'images/enemy-bug.png';
+    this.fieldWidth = fieldWidth;
+    this.x = x;
+    this.y = y;
+    this.speedX = speedX;
+    this.height = 40;
+    this.width = 83;
+    this.player = player;
+  }
+
+  checkCollision() {
+    if (
+      this.x + this.width > this.player.x &&
+      this.player.x + this.player.width > this.x &&
+      this.player.y > this.y - this.height &&
+      this.player.y - this.player.height < this.y
+    ) {
+      this.player.resetPosition();
+    }
+  }
+
+  update(dt) {
+    this.checkCollision();
+
+    if (this.x >= this.fieldWidth) {
+      this.x = 0;
+    }
+
+    this.x += dt * this.speedX;
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+}
+
+class Player {
+  constructor(params) {
```

Comment:
Having an object is a swissknife approach. As a peer developer I would love to know what params (param object key-values) I am expected to pass without reading the further code.

Consider using object destructuring in method declaration.

Also using parameters object on constructor here is a different approach compared to what we had with Enemy constructor above. If we want to add e.g. Gem class for perks in the game what pattern we follow?